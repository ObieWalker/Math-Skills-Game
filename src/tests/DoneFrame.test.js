import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import DoneFrame from '../../src/components/DoneFrame';

let mountedForm, props, wrapper;
const resetGame = jest.fn();
const doneStatus = true


const getComponent = () => {
  if (!mountedForm) {
    props = {
      resetGame,
      doneStatus
    };
    mountedForm = shallow(<DoneFrame {...props} />);
  }
  return mountedForm;
};

describe('DoneFrame Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper.find('div').length).toBe(1);
  });
});
