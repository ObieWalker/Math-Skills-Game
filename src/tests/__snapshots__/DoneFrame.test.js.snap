// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DoneFrame Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DoneFrame
    doneStatus={true}
    resetGame={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          true
        </h2>,
        <button
          className="btn btn-secondary"
          onClick={[MockFunction]}
        >
           Play Again
        </button>,
      ],
      "className": "text-center",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": true,
        },
        "ref": null,
        "rendered": true,
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Play Again",
          "className": "btn btn-secondary",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": " Play Again",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            true
          </h2>,
          <button
            className="btn btn-secondary"
            onClick={[MockFunction]}
          >
             Play Again
          </button>,
        ],
        "className": "text-center",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": true,
          },
          "ref": null,
          "rendered": true,
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Play Again",
            "className": "btn btn-secondary",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": " Play Again",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
